mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1147,14 +1153,75 @@ def _log_metrics_and_artifacts(self):
-                self._log_binary_classifier_artifacts()
-            else:
-                self._log_multiclass_classifier_artifacts()
-            self._log_confusion_matrix()
+        if self.model_type in (_ModelType.CLASSIFIER, _ModelType.REGRESSOR):
+            if self.model_type == _ModelType.CLASSIFIER:
+                if self.is_binomial:
+                    self._log_binary_classifier_artifacts()
+                else:
+                    self._log_multiclass_classifier_artifacts()
+                self._log_confusion_matrix()
+            self._log_model_explainability()
         self._log_metrics()
-        self._log_model_explainability()
+
+    def _log_eval_table(self):
+        metric_prefix = self.evaluator_config.get("metric_prefix", "")
+        if self.dataset.has_targets:
+            data = self.dataset.features_data.assign(
+                **{self.dataset.targets_name or "target": self.y, "outputs": self.y_pred}
+            )
+        else:
+            data = self.dataset.features_data.assign(outputs=self.y_pred)
+        mlflow.log_table(data, artifact_file=f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}")
+
+    def _evaluate_question_answering(self):
+        self._log_eval_table()
+        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
+        self.artifacts[name] = JsonEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+        )
+
+        if self.dataset.has_targets:
+            try:
+                import evaluate
+
+                exact_match = evaluate.load("exact_match")
+            except Exception as e:
+                _logger.warning(
+                    f"Failed to load 'exact_match' metric (error: {e!r}), skipping metric logging."
+                )
+                return
+
+            exact_match = evaluate.load("exact_match")
+            metrics = exact_match.compute(predictions=self.y_pred, references=self.y)
+            self.metrics.update(metrics)
+
+    def _evaluate_text_summarization(self):
+        self._log_eval_table()
+        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
+        self.artifacts[name] = JsonEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+        )
+        if self.dataset.has_targets:
+            try:
+                import evaluate
+
+                rouge = evaluate.load("rouge")
+            except Exception as e:
+                _logger.warning(
+                    f"Failed to load 'rouge' metric (error: {e!r}), skipping metric logging."
+                )
+                return
+
+            rouge = evaluate.load("rouge")
+            metrics = rouge.compute(predictions=self.y_pred, references=self.y)
+            self.metrics.update(metrics)
+
+    def _evaluate_text(self):
+        self._log_eval_table()
+        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
+        self.artifacts[name] = JsonEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+        )
```

Comment:
`targets` are ignored for `model_type='text'`, right? Can we print a warning when users specify `targets` with `model_type='text'`?

Reply:
They are not ignored. They are logged in `eval_results_table.json`.

Reply:
They are also passed to custom metric functions.